### ISSUE TITLE
Empty include.where query broken on 2.0.0

### ISSUE THREAD

**Issue by grahamb** [sequelize/sequelize#3099]
I have the following query:

```
    Promise.all([
        Program.findAll({ order: 'id ASC' }),
        OOS.findAll({
            where: where,
            include: [ { model: Program, where: assignmentFilter } ],
            order: 'oos_number ASC'
        })
    ]).then(function(results) {
        // do something with results
    });
```

This query worked all the way through the 2.0.0 release candidates, including -rc8. It is broken on 2.0.0.

```
App 54187 stdout: Executing (default): SELECT "OOS"."id", "OOS"."oos_number", "OOS"."first_name", "OOS"."last_name", "OOS"."email", "OOS"."address", "OOS"."city", "OOS"."province", "OOS"."country", "OOS"."postal_code", "OOS"."phone", "OOS"."cell_phone", "OOS"."pre_recruited", "OOS"."recruited_by", "OOS"."previous_experience", "OOS"."certifications", "OOS"."dob", "OOS"."notes", "OOS"."import_id", "OOS"."created_at", "OOS"."updated_at", "Programs"."id" AS "Programs.id", "Programs"."name" AS "Programs.name", "Programs"."short_name" AS "Programs.short_name", "Programs"."description" AS "Programs.description", "Programs"."short_description" AS "Programs.short_description", "Programs"."fitness_level" AS "Programs.fitness_level", "Programs"."prerequisites" AS "Programs.prerequisites", "Programs"."knowledge_skills_equipment" AS "Programs.knowledge_skills_equipment", "Programs"."premium_activity" AS "Programs.premium_activity", "Programs"."location" AS "Programs.location", "Programs"."program_periods_available" AS "Programs.program_periods_available", "Programs"."max_participants_per_period" AS "Programs.max_participants_per_period", "Programs"."program_periods_required" AS "Programs.program_periods_required", "Programs"."fee" AS "Programs.fee", "Programs"."hidden" AS "Programs.hidden", "Programs"."oos_required" AS "Programs.oos_required", "Programs"."auto_assign" AS "Programs.auto_assign", "Programs"."include_in_welcome_email" AS "Programs.include_in_welcome_email", "Programs"."created_at" AS "Programs.created_at", "Programs"."updated_at" AS "Programs.updated_at", "Programs"."program_activity_leader_id" AS "Programs.program_activity_leader_id", "Programs.program_oos_assignments"."created_at" AS "Programs.program_oos_assignments.created_at", "Programs.program_oos_assignments"."updated_at" AS "Programs.program_oos_assignments.updated_at", "Programs.program_oos_assignments"."program_id" AS "Programs.program_oos_assignments.program_id", "Programs.program_oos_assignments"."o_o_id" AS "Programs.program_oos_assignments.o_o_id" FROM "OOS" AS "OOS" INNER JOIN ("program_oos_assignments" AS "Programs.program_oos_assignments" INNER JOIN "Programs" AS "Programs" ON "Programs"."id" = "Programs.program_oos_assignments"."program_id") ON "OOS"."id" = "Programs.program_oos_assignments"."o_o_id" AND  ORDER BY oos_number ASC;
App 54187 stdout: { [SequelizeDatabaseError: syntax error at or near "ORDER"]
App 54187 stdout:   name: 'SequelizeDatabaseError',
App 54187 stdout:   message: 'syntax error at or near "ORDER"',
App 54187 stdout:   parent: 
App 54187 stdout:    { [error: syntax error at or near "ORDER"]
App 54187 stdout:      name: 'error',
App 54187 stdout:      length: 89,
App 54187 stdout:      severity: 'ERROR',
App 54187 stdout:      code: '42601',
App 54187 stdout:      detail: undefined,
App 54187 stdout:      hint: undefined,
App 54187 stdout:      position: '2277',
App 54187 stdout:      internalPosition: undefined,
App 54187 stdout:      internalQuery: undefined,
App 54187 stdout:      where: undefined,
App 54187 stdout:      schema: undefined,
App 54187 stdout:      table: undefined,
App 54187 stdout:      column: undefined,
App 54187 stdout:      dataType: undefined,
App 54187 stdout:      constraint: undefined,
App 54187 stdout:      file: 'scan.l',
App 54187 stdout:      line: '1053',
App 54187 stdout:      routine: 'scanner_yyerror',
App 54187 stdout:      sql: 'SELECT "OOS"."id", "OOS"."oos_number", "OOS"."first_name", "OOS"."last_name", "OOS"."email", "OOS"."address", "OOS"."city", "OOS"."province", "OOS"."country", "OOS"."postal_code", "OOS"."phone", "OOS"."cell_phone", "OOS"."pre_recruited", "OOS"."recruited_by", "OOS"."previous_experience", "OOS"."certifications", "OOS"."dob", "OOS"."notes", "OOS"."import_id", "OOS"."created_at", "OOS"."updated_at", "Programs"."id" AS "Programs.id", "Programs"."name" AS "Programs.name", "Programs"."short_name" AS "Programs.short_name", "Programs"."description" AS "Programs.description", "Programs"."short_description" AS "Programs.short_description", "Programs"."fitness_level" AS "Programs.fitness_level", "Programs"."prerequisites" AS "Programs.prerequisites", "Programs"."knowledge_skills_equipment" AS "Programs.knowledge_skills_equipment", "Programs"."premium_activity" AS "Programs.premium_activity", "Programs"."location" AS "Programs.location", "Programs"."program_periods_available" AS "Programs.program_periods_available", "Programs"."max_participants_per_period" AS "Programs.max_participants_per_period", "Programs"."program_periods_required" AS "Programs.program_periods_required", "Programs"."fee" AS "Programs.fee", "Programs"."hidden" AS "Programs.hidden", "Programs"."oos_required" AS "Programs.oos_required", "Programs"."auto_assign" AS "Programs.auto_assign", "Programs"."include_in_welcome_email" AS "Programs.include_in_welcome_email", "Programs"."created_at" AS "Programs.created_at", "Programs"."updated_at" AS "Programs.updated_at", "Programs"."program_activity_leader_id" AS "Programs.program_activity_leader_id", "Programs.program_oos_assignments"."created_at" AS "Programs.program_oos_assignments.created_at", "Programs.program_oos_assignments"."updated_at" AS "Programs.program_oos_assignments.updated_at", "Programs.program_oos_assignments"."program_id" AS "Programs.program_oos_assignments.program_id", "Programs.program_oos_assignments"."o_o_id" AS "Programs.program_oos_assignments.o_o_id" FROM "OOS" AS "OOS" INNER JOIN ("program_oos_assignments" AS "Programs.program_oos_assignments" INNER JOIN "Programs" AS "Programs" ON "Programs"."id" = "Programs.program_oos_assignments"."program_id") ON "OOS"."id" = "Programs.program_oos_assignments"."o_o_id" AND  ORDER BY oos_number ASC;' },
App 54187 stdout:   original: 
App 54187 stdout:    { [error: syntax error at or near "ORDER"]
App 54187 stdout:      name: 'error',
App 54187 stdout:      length: 89,
App 54187 stdout:      severity: 'ERROR',
App 54187 stdout:      code: '42601',
App 54187 stdout:      detail: undefined,
App 54187 stdout:      hint: undefined,
App 54187 stdout:      position: '2277',
App 54187 stdout:      internalPosition: undefined,
App 54187 stdout:      internalQuery: undefined,
App 54187 stdout:      where: undefined,
App 54187 stdout:      schema: undefined,
App 54187 stdout:      table: undefined,
App 54187 stdout:      column: undefined,
App 54187 stdout:      dataType: undefined,
App 54187 stdout:      constraint: undefined,
App 54187 stdout:      file: 'scan.l',
App 54187 stdout:      line: '1053',
App 54187 stdout:      routine: 'scanner_yyerror',
App 54187 stdout:      sql: 'SELECT "OOS"."id", "OOS"."oos_number", "OOS"."first_name", "OOS"."last_name", "OOS"."email", "OOS"."address", "OOS"."city", "OOS"."province", "OOS"."country", "OOS"."postal_code", "OOS"."phone", "OOS"."cell_phone", "OOS"."pre_recruited", "OOS"."recruited_by", "OOS"."previous_experience", "OOS"."certifications", "OOS"."dob", "OOS"."notes", "OOS"."import_id", "OOS"."created_at", "OOS"."updated_at", "Programs"."id" AS "Programs.id", "Programs"."name" AS "Programs.name", "Programs"."short_name" AS "Programs.short_name", "Programs"."description" AS "Programs.description", "Programs"."short_description" AS "Programs.short_description", "Programs"."fitness_level" AS "Programs.fitness_level", "Programs"."prerequisites" AS "Programs.prerequisites", "Programs"."knowledge_skills_equipment" AS "Programs.knowledge_skills_equipment", "Programs"."premium_activity" AS "Programs.premium_activity", "Programs"."location" AS "Programs.location", "Programs"."program_periods_available" AS "Programs.program_periods_available", "Programs"."max_participants_per_period" AS "Programs.max_participants_per_period", "Programs"."program_periods_required" AS "Programs.program_periods_required", "Programs"."fee" AS "Programs.fee", "Programs"."hidden" AS "Programs.hidden", "Programs"."oos_required" AS "Programs.oos_required", "Programs"."auto_assign" AS "Programs.auto_assign", "Programs"."include_in_welcome_email" AS "Programs.include_in_welcome_email", "Programs"."created_at" AS "Programs.created_at", "Programs"."updated_at" AS "Programs.updated_at", "Programs"."program_activity_leader_id" AS "Programs.program_activity_leader_id", "Programs.program_oos_assignments"."created_at" AS "Programs.program_oos_assignments.created_at", "Programs.program_oos_assignments"."updated_at" AS "Programs.program_oos_assignments.updated_at", "Programs.program_oos_assignments"."program_id" AS "Programs.program_oos_assignments.program_id", "Programs.program_oos_assignments"."o_o_id" AS "Programs.program_oos_assignments.o_o_id" FROM "OOS" AS "OOS" INNER JOIN ("program_oos_assignments" AS "Programs.program_oos_assignments" INNER JOIN "Programs" AS "Programs" ON "Programs"."id" = "Programs.program_oos_assignments"."program_id") ON "OOS"."id" = "Programs.program_oos_assignments"."o_o_id" AND  ORDER BY oos_number ASC;' },
```

The exact same code generates the following query on 2.0.0-rc8:

```
App 55705 stdout: Executing (default): SELECT "OOS"."id", "OOS"."oos_number", "OOS"."first_name", 
"OOS"."last_name", "OOS"."email", "OOS"."address", "OOS"."city", "OOS"."province", "OOS"."country", 
"OOS"."postal_code", "OOS"."phone", "OOS"."cell_phone", "OOS"."pre_recruited", "OOS"."recruited_by", 
"OOS"."previous_experience", "OOS"."certifications", "OOS"."dob", "OOS"."notes", "OOS"."import_id", 
"OOS"."created_at", "OOS"."updated_at", "Programs"."id" AS "Programs.id", "Programs"."name" AS 
"Programs.name", "Programs"."short_name" AS "Programs.short_name", "Programs"."description" AS 
"Programs.description", "Programs"."short_description" AS "Programs.short_description", 
"Programs"."fitness_level" AS "Programs.fitness_level", "Programs"."prerequisites" AS 
"Programs.prerequisites", "Programs"."knowledge_skills_equipment" AS 
"Programs.knowledge_skills_equipment", "Programs"."premium_activity" AS 
"Programs.premium_activity", "Programs"."location" AS "Programs.location", 
"Programs"."program_periods_available" AS "Programs.program_periods_available", 
"Programs"."max_participants_per_period" AS "Programs.max_participants_per_period", 
"Programs"."program_periods_required" AS "Programs.program_periods_required", "Programs"."fee" AS 
"Programs.fee", "Programs"."hidden" AS "Programs.hidden", "Programs"."oos_required" AS 
"Programs.oos_required", "Programs"."auto_assign" AS "Programs.auto_assign", 
"Programs"."include_in_welcome_email" AS "Programs.include_in_welcome_email", 
"Programs"."created_at" AS "Programs.created_at", "Programs"."updated_at" AS 
"Programs.updated_at", "Programs"."program_activity_leader_id" AS 
"Programs.program_activity_leader_id", "Programs.program_oos_assignments"."created_at" AS 
"Programs.program_oos_assignments.created_at", "Programs.program_oos_assignments"."updated_at"
 AS "Programs.program_oos_assignments.updated_at", 
"Programs.program_oos_assignments"."program_id" AS 
"Programs.program_oos_assignments.program_id", "Programs.program_oos_assignments"."o_o_id" AS
 "Programs.program_oos_assignments.o_o_id" FROM "OOS" AS "OOS" INNER JOIN 
("program_oos_assignments" AS "Programs.program_oos_assignments" INNER JOIN "Programs" AS 
"Programs" ON "Programs"."id" = "Programs.program_oos_assignments"."program_id") ON "OOS"."id" = 
"Programs.program_oos_assignments"."o_o_id" AND 1=1 WHERE 1=1 ORDER BY oos_number ASC;
```

The difference seems to be right at the end, at the `ORDER BY` statement:

2.0.0-rc8: ` AND 1=1 WHERE 1=1 ORDER BY oos_number ASC;`
2.0.0: `AND  ORDER BY oos_number ASC;`

**Comment by mickhansen** [comment]
Not an issue with order but rather an issue with `where` on include it seems.

**Comment by mickhansen** [comment]
Fix published in `sequelize@2.0.1`
